src/gui/ScrollHelpers.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * ScrollHelpers.cpp - helper functions for wheel events
+ *
+ * Copyright (c) 2023 Alex <allejok96/gmail>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "ScrollHelpers.h"
+
+#include <QWheelEvent>
+
+
+namespace lmms {
+
+
+bool ignoreScroll(const Qt::Orientation orientation, QWheelEvent* event)
+{
+	bool hasOtherOrientation = orientation == Qt::Horizontal ? event->angleDelta().y() : event->angleDelta().x();
+	event->setAccepted(hasOtherOrientation);
+	return !hasOtherOrientation;
+}
+
+
+
+
+// TODO: is there a good way to prevent calling this method multiple times with the same event?
```

Comment:
Where are we calling it multiple times? I don't seem to see any.

Reply:
For diagonal scroll events, Qt 5 generates two events(one for vertical, the other for horizontal) for Qt 4 compatibility. AFAIK, this isn't the case in Qt 6.

Reply:
Thanks for the insight @PhysSong, I was wondering about that. This is not a problem atm, but I was thinking about fool proofing. I tried to make it clear in the doc string, but you know.... people. If called improperly the counter will be double increased and it will be hard to detect.